test/es-module/test-cjs-esm-warn.mjs

```diff
@@ -0,0 +1,64 @@
+import { mustCall } from '../common/index.mjs';
+import * as fixtures from '../common/fixtures.mjs';
+import assert from 'node:assert';
+import path from 'node:path';
+import { execPath } from 'node:process';
+
+import spawn from './helper.spawnAsPromised.mjs';
+
+const requiringCjsAsEsm = path.resolve(fixtures.path('/es-modules/cjs-esm.js'));
+const requiringEsm = path.resolve(fixtures.path('/es-modules/cjs-esm-esm.js'));
+const pjson = path.resolve(
+  fixtures.path('/es-modules/package-type-module/package.json')
+);
+
+{
+  const required = path.resolve(
+    fixtures.path('/es-modules/package-type-module/cjs.js')
+  );
+  const basename = 'cjs.js';
+  spawn(execPath, [requiringCjsAsEsm])
+    .then(mustCall(({ code, signal, stderr }) => {
```

Comment:
Using `.then` is unfortunate, why not instead use async/await and the `node:test` module to handle the parallelisation?

Reply:
Say what now?

Reply:
Instead of:

```js
{
  const someVar = 'value';
  spawn(/* args */)
    .then(common.mustCall((spawnResult) => {
      // ...
    });
}
{
  const someVar = 'other value';
  spawn(/* args */)
    .then(common.mustCall((spawnResult) => {
      // ...
    });
}
```

We could write:

```js
test(async () => {
  const someVar = 'value';
  const spawnResult = await spawn(/* args */)
  // ...
});

test(async () => {
  const someVar = 'other value';
  const spawnResult = await spawn(/* args */)
  // ...
});

Reply:
is this what you had in mind? 04c4649

I'm not sure I got the concurrency right. I'm also not sure why `concurrency` accepts an integer—I think I would never want to control the number of concurrent tests, merely "yes" I want concurrency, or "no" I don't.